src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs

```diff
@@ -252,6 +252,24 @@ static async IAsyncEnumerable<int> GetNumbersAsync()
             }
         }
 
+        [Fact]
+        public async Task RegressionTest_ExceptionOnFirstMoveNextShouldNotFlushBuffer()
+        {
+            // Regression test for https://github.com/dotnet/aspnetcore/issues/36977
+            using var stream = new MemoryStream();
+            await Assert.ThrowsAsync<NotImplementedException>(async () => await JsonSerializer.SerializeAsync(stream, new { Data = GetFailingAsyncEnumerable() }));
+            Assert.Equal(0, stream.Length);
+
+            static async IAsyncEnumerable<int> GetFailingAsyncEnumerable()
+            {
+                await Task.Yield();
+                throw new NotImplementedException();
+#pragma warning disable CS0162 // Unreachable code detected
+                yield break;
```

Comment:
what's this for? won't it compile and execute the test scenario just fine without it?

Reply:
A yield expression is necessary for async enumerables to compile. But you still need to suppress the unreachable code detection warning.